lib/services/role.go

```diff
@@ -2015,10 +2015,16 @@ func (set RoleSet) checkAccess(r AccessCheckable, mfa AccessMFAParams, matchers
 	additionalDeniedMessage := ""
 
 	var getRoleLabels func(types.Role, types.RoleConditionType) types.Labels
+	resourceHasLabels := true
+	getEmptyLabels := func(r types.Role, rct types.RoleConditionType) types.Labels { return types.Labels{} }
+
 	switch r.GetKind() {
 	case types.KindDatabase:
 		getRoleLabels = types.Role.GetDatabaseLabels
 		additionalDeniedMessage = "Confirm database user and name."
+	case types.KindDatabaseService:
+		resourceHasLabels = false
+		getRoleLabels = getEmptyLabels
```

Comment:
This is a little confusing tbh. Why the special case for database service?

Reply:
I agree, it's not very clear.

In order to re-use the ListResources method for DatabaseServices, we end here.
The other resources have associated labels in the role
KindDatabase: `db_labels`
KindApp: `app_labels`
KindNode: 'node_labels`
And so on.

However, DatabaseServices doesn't have an associated `<x>_labels` field in the Role.Allow spec.
So, to be able to re-use the ListResources, I had to add a little branch on the code to allow for that case.

There was a little discussion about this in the `master`'s PR
https://github.com/gravitational/teleport/pull/19469#discussion_r1052271379

Reply:
@marcoandredinis I guess my point is, why not just add `GetDatabaseServiceLabels` to the new resource and use it here for consistency without introducing extra flags, even it if will always return empty? If you look below, WindowsDesktop/WindowsDesktopService do this, basically similar situation, no?

I'd just really like to avoid extra scenarios in RBAC checks, opens up a way for errors.

Reply:
The WindowsDesktop and WindowsDesktopService use the `windows_desktop_labels` set of labels.
So it's actually treated as a single resource. You cannot set different labels for WindowsDesktop and WindowsDesktopService.

I'm not sure about grouping applying the same logic here: have the `db_labels` as the source for the DatabaseServices allowed labels.
I see them as two separate resources.

Is your suggestion to use `db_labels` as the source for allowing access to DatabaseServices?
Or should we create a `GetDatabaseServiceLabels` that always returns `'*': '*'` ? ( I don't think returning an empty set of labels works)

Reply:
After talking to Roman we concluded that this is hackish at best and will instead create a `db_services_labels` as Role Allow/Deny Option and set it to 
A new PR will be created and once approved, cherry-pick it to this backport PR